src/gen_rules.ml

```diff
@@ -704,11 +728,17 @@ module Gen(P : Install_rules.Params) = struct
        match SC.stanzas_in sctx ~dir with
        | Some x -> gen_rules x
        | None ->
-         if components <> [] &&
-            Option.is_none
-              (File_tree.find_dir (SC.file_tree sctx)
-                 (Path.drop_build_context_exn dir)) then
-           SC.load_dir sctx ~dir:(Path.parent_exn dir));
+         if components <> [] then begin
+           match File_tree.find_dir (SC.file_tree sctx)
+                   (Path.drop_build_context_exn dir) with
+           | None ->
+             SC.load_dir sctx ~dir:(Path.parent_exn dir)
+           | Some _ ->
+             match Dir_contents.kind (Dir_contents.get sctx ~dir) with
+             | Group_part root ->
+               SC.load_dir sctx ~dir:(Dir_contents.dir root)
```

Comment:
How come we only do this in the case when there are no stanzas in `dir`? Seems like we should let the paren that is part of multi dir lib regardless

Reply:
It was done in the other `gen_rules` function. However, it's true that this code wasn't very clear and was starting to become spaghetti code, so I refactored it